### PR TITLE
Igor lig 801 check embeddings fails with masked

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

- Make sure we remove all special columns such as `masked`, `selected` (also if they appear earlier)
- Add new test case

### CODE REVIEW

**Comment by philippmwirth** [comment]
Looks like you didn't checkout from `master` but from your local branch instead (this PR contains all the changes of #726 as far as I understand).

**Review by MalteEbner — COMMENTED** [review]
Could we remove the `remove_additional_columns` argument and handle as if it is always true?  Or is there a case where we don't want to remove additional columns?

**Comment by IgorSusmelj** [comment]
> Could we remove the `remove_additional_columns` argument and handle as if it is always true? Or is there a case where we don't want to remove additional columns?

We can look into this in a follow up issue

File: lightly/utils/io.py [hunk 1]
```diff
@@ -98,12 +101,25 @@ def check_embeddings(path: str, remove_additional_columns: bool=False):
...
             for row in reader:
-                row = row[:header_labels_idx+1]
-                new_rows.append(row)
+                # apply mask to only use filenames, embedding_ or labels
+                new_rows.append(list(compress(row, col_mask)))
+
         with open(path, 'w', newline='') as csv_file:
             writer = csv.writer(csv_file, delimiter=',')
             writer.writerows(new_rows)
+            print(new_rows)
```
Comment (MalteEbner) [inline lightly/utils/io.py:122]: Do we need this print?

File: tests/utils/test_io.py [hunk 2]
```diff
@@ -100,4 +100,26 @@ def test_embeddings_extra_rows(self):
             for row_read, row_original in zip(csv_reader, rows):
                 self.assertListEqual(row_read, row_original[:-2])
 
+    def test_embeddings_extra_rows_special_order(self):
```
Comment (MalteEbner) [inline tests/utils/test_io.py:103]: Great unittest!

**Review by MalteEbner — APPROVED** [review]
This looks great! I would only remove the print.

File: lightly/utils/io.py (hunk 1 above)
Comment (IgorSusmelj) [reply lightly/utils/io.py:122]: Nooooo, good catch!

**Pull request merged**